apps/twap/src/app/plugins/orderbook.ts

```diff
@@ -0,0 +1,110 @@
+import { FastifyInstance } from 'fastify';
+import fp from 'fastify-plugin';
+
+import { Settlement } from '../orderbook/settlement';
+import { DataSource, Repository } from 'typeorm';
+import { Trade } from '../orderbook/trade';
+import { Order } from '../orderbook/order';
+import { SupportedChainId } from '@cowprotocol/cow-sdk';
+import { getApiBaseUrl } from '../utils/getApiBaseUrl';
+import { ExecutionInfo } from '../types/order';
+
+const DEFAULT_TWAP_EXECUTION_INFO = {
+  executedSellAmount: BigInt(0),
+  executedBuyAmount: BigInt(0),
+  executedFeeAmount: BigInt(0),
+};
+
+function getExecutionInfoFactory(dataSource: DataSource, apiBaseUrl: string) {
+  return async function getExecutionInfo(
+    orderId: string,
+    partIds: string[]
+  ): Promise<ExecutionInfo> {
+    // TODO: We need to create a query here to derive the execution info from the
+    // read only db.
+    // However, because staging db does not have any of the data we need, we can't rely on this.
+    // Based on: https://github.com/cowprotocol/services/blob/main/crates/database/src/orders.rs#L483-L549
+    // Other options are: Subgraph or watch for the settlement event
+    try {
+      const orders: ExecutionInfo[] = await Promise.all(
+        partIds.map((partId) =>
+          fetch(`${apiBaseUrl}/orders/${partId}`)
+            .then((res) => res.json())
+            .then((json) => ({
+              executedBuyAmount: BigInt(json.executedBuyAmount),
+              executedSellAmount: BigInt(json.executedSellAmount),
+              executedFeeAmount: BigInt(json.executedFeeAmount),
```

Comment:
Are you sure this is what you are looking for?

For non-swap orders, this will always be 0 AFAICT.

In the SDK we have one transformation step to fill this the way one would expect 

https://github.com/cowprotocol/cow-sdk/blob/2e5cc9df8cad563ae3f51a2aedf6efe22dbc16e6/src/order-book/transformOrder.ts#L16

See the example TWAP part order in the previous comment
<img width="398" alt="Screenshot 2023-07-14 at 16 26 08" src="https://github.com/cowprotocol/cow-web-services/assets/43217/df46074f-2730-459a-89c1-a5d2e295c7e0">

Reply:
Leandro is right. But, for TWAP frontend we don't need this value. For now we can just skip it at all

Reply:
However, in general we should use the transformation here

Reply:
Will update this - thank you :+1: